netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+                        // no winner, skip it.
+                        // don't add to conflict list and just use the existing item for future conflicts.
+                        continue;
+                    }
+
+                    if (winner != existingItem)
+                    {
+                        // replace existing item and add it to conflict list.
+                        targetPathToItem[targetPath] = item;
+                        conflicts.Add(existingItem);
+                    }
+                    else
+                    {
+                        // no need to replace item, just add new item to conflict list.
+                        conflicts.Add(item);
+                    }
+                }
+                else
+                {
+                    targetPathToItem[targetPath] = item;
+                }
+            }
+
+            return RemoveConflicts(items, conflicts);
+        }
+
+        private Version GetFileVersion(string sourcePath)
+        {
+            var fvi = FileVersionInfo.GetVersionInfo(sourcePath);
+
+            if (fvi != null)
+            {
+                return new Version(fvi.FileMajorPart, fvi.FileMinorPart, fvi.FileBuildPart, fvi.FilePrivatePart);
+            }
+
+            return null;
+        }
+
+        private int GetPackageRank(ITaskItem item)
+        {
+            int rank = int.MaxValue;
+
+            // NuGet 3
+            var packageId = item.GetMetadata("NuGetPackageId");
+
+            if (String.IsNullOrWhiteSpace(packageId))
+            {
+                // NuGet 4
+                packageId = item.GetMetadata("ParentPackage");
+
+                var versionSeperatorIndex = packageId.IndexOf('/');
+
+                if (versionSeperatorIndex != -1)
+                {
+                    packageId = packageId.Substring(0, versionSeperatorIndex);
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(packageId) && PreferredPackages != null)
+            {
+                EnsurePackageRanks();
+
+                packageRanks.TryGetValue(packageId, out rank);
+            }
+
+            return rank;
+        }
+
+        private static string GetSourcePath(ITaskItem item)
+        {
+            var sourcePath = item.GetMetadata("HintPath");
+
+            if (String.IsNullOrWhiteSpace(sourcePath))
+            {
+                sourcePath = item.ItemSpec;
+            }
+
+            return sourcePath;
+        }
+
+        static readonly string[] s_targetPathMetadata = new[] { "TargetPath", "Path" };
+        private static string GetTargetPath(ITaskItem item)
+        {
+            // first use TargetPath, then Path, then fallback to filename+extension alone
+            foreach (var metadata in s_targetPathMetadata)
+            {
+                var value = item.GetMetadata(metadata);
+
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    return value.Replace('\\', '/');
```

Comment:
Is this a your version of a cheap normalize? We might need to use a better normalize at some point.